subcommand/mesh-init/dataplane-monitor.go

```diff
@@ -0,0 +1,99 @@
+package meshinit
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul-ecs/config"
+	"github.com/hashicorp/go-hclog"
+)
+
+type dataplaneMonitor struct {
+	ctx context.Context
+	log hclog.Logger
+
+	doneCh chan struct{}
+}
+
+func newDataplaneMonitor(ctx context.Context, logger hclog.Logger) *dataplaneMonitor {
+	return &dataplaneMonitor{
+		ctx:    ctx,
+		log:    logger,
+		doneCh: make(chan struct{}, 1),
```

Comment:
nit: I think this channel can be unbuffered

```suggestion
		doneCh: make(chan struct{}),
```